content/blog/2018/egress-monitoring-access-control.md

```diff
@@ -0,0 +1,546 @@
+          variables:
+            destination: request.host | "unknown"
+            path: request.path | "unknown"
+            source: source.labels["app"] | source.service | "unknown"
+            sourceNamespace: source.namespace | "unknown"
+            user: source.user | "unknown"
+            responseCode: response.code | 0
+            responseSize: response.size | 0
+          monitored_resource_type: '"UNSPECIFIED"'
+        ---
+        # Handler for error egress access entries
+        apiVersion: "config.istio.io/v1alpha2"
+        kind: stdio
+        metadata:
+          name: egress-error-logger
+          namespace: istio-system
+        spec:
+         severity_levels:
+           info: 2 # output log level as error
+         outputAsJson: true
+        ---
+        # Rule to handle access to *.cnn.com/politics
+        apiVersion: "config.istio.io/v1alpha2"
+        kind: rule
+        metadata:
+          name: handle-politics
+          namespace: istio-system
+        spec:
+          match: request.host.endsWith("cnn.com") && request.path.startsWith("/politics")
+          actions:
+          - handler: egress-error-logger.stdio
+            instances:
+            - egress-access.logentry
+        ---
+        # Handler for info egress access entries
+        apiVersion: "config.istio.io/v1alpha2"
+        kind: stdio
+        metadata:
+          name: egress-access-logger
+          namespace: istio-system
+        spec:
+          severity_levels:
+            info: 0 # output log level as info
+          outputAsJson: true
+        ---
+        # Rule to handle access to *.cnn.com
+        apiVersion: "config.istio.io/v1alpha2"
+        kind: rule
+        metadata:
+          name: handle-cnn-access
+          namespace: istio-system
+        spec:
+          match: request.host.endsWith(".cnn.com")
+          actions:
+          - handler: egress-access-logger.stdio
+            instances:
+              - egress-access.logentry
+        EOF
+    ```
+
+1.  Let's send three HTTP requests to _cnn.com_, to [edition.cnn.com/politics](https://edition.cnn.com/politics), [edition.cnn.com/sport](https://edition.cnn.com/sport) and [edition.cnn.com/health](https://edition.cnn.com/health). All three should return _200 OK_.
+
+    ```command
+    $ kubectl exec -it $SOURCE_POD -c sleep -- bash -c 'curl -sL -o /dev/null -w "%{http_code}\n" http://edition.cnn.com/politics; curl -sL -o /dev/null -w "%{http_code}\n" http://edition.cnn.com/sport; curl -sL -o /dev/null -w "%{http_code}\n" http://edition.cnn.com/health'
+    200
+    200
+    200
+    ```
+
+1.  Let's query the Mixer log and see that the information about the requests appear in the log:
+    ```command
+    $ kubectl -n istio-system logs $(kubectl -n istio-system get pods -l istio-mixer-type=telemetry -o jsonpath='{.items[0].metadata.name}') mixer | grep egress-access | grep cnn | tail -4
+    ```
+
+    The output should be similar to the following:
+
+    ```plain
```

Comment:
plain -> json

Alternatively, you can combine this with the previous block using:

    command-output-as-json